lua/lspkind/init.lua

```diff
@@ -1,185 +1,303 @@
-local lspkind = {}
+local M = {}
+local H = {}
 local fmt = string.format
 
-local kind_presets = {
-  default = {
--- if you change or add symbol here
--- replace corresponding line in readme
-    Text = "",
-    Method = "",
-    Function = "",
-    Constructor = "",
-    Field = "ﰠ",
-    Variable = "",
-    Class = "ﴯ",
-    Interface = "",
-    Module = "",
-    Property = "ﰠ",
-    Unit = "塞",
-    Value = "",
-    Enum = "",
-    Keyword = "",
-    Snippet = "",
-    Color = "",
-    File = "",
-    Reference = "",
-    Folder = "",
-    EnumMember = "",
-    Constant = "",
-    Struct = "פּ",
-    Event = "",
-    Operator = "",
-    TypeParameter = ""
+local kinds = {
+  -- Kind items order
+  order = {
```

Comment:
Why on earth do we need `order`? I think this is not necessary at all -- we can just lookup from a table. Assigning LSP kinds to numbers in a somewhat random order doesn't sound like a good idea.

Reply:
It is not a random order of numbers, this is properly implemented in the core, if you run nvim --clean -c 'lua vim.pretty_print(vim.lsp.protocol.CompletionItemKind)' you will see the names assigned to the numbers
Symbol tables are indexed by number, eg table[1] = 'SomeKind'
In the case of the code, it is so that the functions do not get lost and return a symbol in the wrong position.
But I think there must be some way to get it directly from the core and be sure that it didn't return to the wrong position

Reply:
If that's the case why are you hard-coding it? That's even more strange. I still believe it is not necessary -- please see comments around L238.